zeebe/exporters/operate-exporter/src/main/java/io/camunda/zeebe/operate/exporter/OperateElasticsearchExporter.java

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.zeebe.operate.exporter;
+
+import static io.camunda.zeebe.protocol.record.ValueType.PROCESS;
+
+import co.elastic.clients.elasticsearch.ElasticsearchClient;
+import io.camunda.operate.exceptions.PersistenceException;
+import io.camunda.operate.schema.indices.ProcessIndex;
+import io.camunda.operate.store.elasticsearch.NewElasticsearchBatchRequest;
+import io.camunda.zeebe.exporter.api.Exporter;
+import io.camunda.zeebe.exporter.api.context.Context;
+import io.camunda.zeebe.exporter.api.context.Controller;
+import io.camunda.zeebe.operate.exporter.handlers.ProcessHandler;
+import io.camunda.zeebe.protocol.record.Record;
+import io.camunda.zeebe.protocol.record.RecordType;
+import io.camunda.zeebe.protocol.record.ValueType;
+import java.time.Duration;
+import java.util.List;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class OperateElasticsearchExporter implements Exporter {
```

Comment:
Suggestion: This class is tightly coupled to its various internal components (client, writer, controller, etc). Instead of private create methods within this class, create separate factory/builder classes that are passed into the constructor and delegate the logic there. This allows for better testing of this particular class in isolation.